### ISSUE TITLE
Notice: Array to string conversion

### ISSUE THREAD

**Issue by cynthiawoodard** [CiviCooP/org.civicoop.civirules#53]
I am trying to log activities when a contact is added and removed from certain groups.

After adding the first rule to log an activity when someone is added to a certain group, I got this error when assigning a contact to that group via Edit Contact, then saving. I get sent to the View Contact screen and see this error:

Notice: Array to string conversion in CRM_CivirulesActions_Generic_Api->executeApiAction() (line 73 of /home/***/public_html/sites/default/files/civicrm/ext/org.civicoop.civirules-master/CRM/CivirulesActions/Generic/Api.php).

The group did not end up getting added and neither did the activity.

If I try adding the group via the View screen, the group gets added fine, and there is no error, but it doesn't trigger the rule it seems. No activity gets added.

Drupal 7.41
Civicrm 4.6.10
Civirules: downloaded from github on 12/16/2015

Let me know if there is any more information needed :) And thank you for an amazing extension. I'm just starting to set up for my organization, but I can already tell this is going to be useful.

![rule](https://cloud.githubusercontent.com/assets/10617817/11890348/766e4e1a-a521-11e5-852a-3f3ad0500b5e.jpg)

![error](https://cloud.githubusercontent.com/assets/10617817/11890536/6819115e-a523-11e5-90b1-df7dcc2127b2.jpg)

**Comment by ErikHommel** [comment]
Have you tried the latest version? We had an earlier bug with activities that has now been solved #30 and that might solve yours too?

**Comment by cynthiawoodard** [comment]
I downloaded from github two days ago, so it should probably have that fix in it?

**Comment by cynthiawoodard** [comment]
FYI, I downgraded to CiviCRM 4.6.10 (fresh install) and I'm getting the same error still. I thought at first maybe it was related to me having Individual subtypes of Adult and Child, but error happened still even after I tried getting rid of them.

**Comment by elcapo** [comment]
I found the same problem (CiviRules 1.2 over CiviCRM 4.6.10) trying to use the same action (add activity to contact). I'm still not sure about what causes it but I've seen a value that I believe is related with the issue at **CRM_CivirulesActions_Activity_Add**.

In particular, I've seen that, after the execution of **alterApiParameters**, the value of the parameter **assignee_contact_id** is equal to an array with an empty string (not empty):

```
$params['assignee_contact_id'] = ['']
```

Source: https://github.com/CiviCooP/org.civicoop.civirules/blob/master/CRM/CivirulesActions/Activity/Add.php#L27

I don't have time to look at it today, so I'm writing the message here as a reminder.

**Comment by jaapjansma** [comment]
I get the same the notice however Civirules seems to behave correctly. I have added a fix to remove the notices.

**Comment by cynthiawoodard** [comment]
Will removing the error message fix my issue with the group and activity not getting logged correctly? I'm a bit confused by this response.

**Comment by cynthiawoodard** [comment]
FYI, elcapo's changes fix my problem. Activities are now being added correctly. Thank you!

https://github.com/CiviCooP/org.civicoop.civirules/pull/54
